### PR TITLE
TextArea: Customizable listbox colors with more readable SELECTED color

### PR DESCRIPTION
**Pull request by bsdcode**
## Description
Make the colors inside listboxes (e.g. movie or character sound selection) customizable via colors.2da. This also changes the color for SELECTED items from green to orange, which is the correct color used in the original BG1 and which is also better readable.

Green SELECTED color:
movie_gemrb0
Orange SELECTED color:
movie_gemrb1
Original SELECTED color:
movie_wine

I haven't found the exact color value for the orange with NI, this is a guessed value which looks similar enough. If the exact color value is known, also for the other games, then we can change this color again.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I used the same coding style as the surrounding code
- [x] I have tested the proposed changes
- [ ] I extended the documentation, if necessary
- [ ] The proposed change builds also on our build bots (check after submission)

### CODE REVIEW

File: gemrb/GUIScripts/MessageWindow.py [hunk 1]
```diff
@@ -40,8 +40,8 @@ def OnLoad():
 	TMessageTA.SetFlags(IE_GUI_TEXTAREA_AUTOSCROLL|IE_GUI_TEXTAREA_HISTORY)
 	TMessageTA.SetResizeFlags(IE_GUI_VIEW_RESIZE_ALL)
 	TMessageTA.AddAlias("MsgSys", 0)
-	TMessageTA.SetColor({'r' : 255, 'g' : 0, 'b' : 0}, TA_COLOR_OPTIONS)
-	TMessageTA.SetColor({'r' : 255, 'g' : 255, 'b' : 255}, TA_COLOR_HOVER)
+	TMessageTA.SetColor(ColorRed, TA_COLOR_OPTIONS)
+	TMessageTA.SetColor(ColorWhite, TA_COLOR_HOVER)
```
Comment (lynxlynxlynx) [inline gemrb/GUIScripts/MessageWindow.py:44]: I bet these should be the whitish kind.

**Comment by lynxlynxlynx** [comment]
3ce24df60 is too bold. If you don't know how the other games do it, best keep their color the same. It's a very specific green, so it's surely used somewhere. I can see that all bg1 lists use the "orange" hover though.

**Comment by bradallred** [comment]
Some resources that might be helpful in locating the colors: https://gibberlings3.github.io/iesdp/appendices/colourgradients.htm

If the EE's use the same color you can glean an awful lot of UI information in NI by looking at the GLSL, GUI, MENU, and SQL data.

File: gemrb/GUIScripts/MessageWindow.py (hunk 1 above)
Comment (bsdcode) [reply gemrb/GUIScripts/MessageWindow.py]: As with the other listbox colors I can't find reference for it in the CHU via NI. I am trusting your judgement here and changed it accordingly.

**Comment by bsdcode** [comment]
> [3ce24df](https://github.com/gemrb/gemrb/commit/3ce24df6062fe969d5f6f22e5e8e747725e2be43) is too bold. If you don't know how the other games do it, best keep their color the same. It's a very specific green, so it's surely used somewhere. I can see that all bg1 lists use the "orange" hover though.

Ok, I changed it back to the green value for non-BG1 games. Judging by some videos for BG2 I think that at least BG2 has also an orange color for the SELECTED state. But without further research it's probably best to revert it for BG2 too.

> Some resources that might be helpful in locating the colors: https://gibberlings3.github.io/iesdp/appendices/colourgradients.htm
> 
> If the EE's use the same color you can glean an awful lot of UI information in NI by looking at the GLSL, GUI, MENU, and SQL data.

I tried all orange-y looking colors from the table, but none of these seem to fit the orange color used it BG1. When I finally get around buying all the other games I can return to this topic and probably research the exact color values. For now I think the approximated orange is close enough.

File: gemrb/GUIScripts/MessageWindow.py (hunk 1 above)
Comment (lynxlynxlynx) [reply gemrb/GUIScripts/MessageWindow.py]: You can always use a color picker for an estimate, but this was just going off of the fact that pure colors are rare.

Comment (bsdcode) [reply gemrb/GUIScripts/MessageWindow.py]: This is a good idea! But even with a color picker I get no better answers. The white values I get with a picker for the HOVER color are not unambiguous and span a rather wide range of values. In the end I resorted to plain old screenshots ;)

GemRB with `ColorWhitish`
gemrb

Original
wine

GemRB with `ColorWhite`
gemrb1

I'd say `ColorWhite` is closer to the original than `ColorWhitish`. Without doing deeper research we probably won't get the correct original value now. I will revert the color back to `ColorWhite`. Getting the correct original colors is a topic for a differnt issue/PR.

**Comment by bsdcode** [comment]
I streamlined/dropped the last commits. Only the SELECTED color for BG1 is changed now.

**Comment by lynxlynxlynx** [comment]
Thanks!

**Comment by bradallred** [comment]
> I tried all orange-y looking colors from the table, but none of these seem to fit the orange color used it BG1

you can try #ffe286. I found this in BGEE.LUA
```lua
fontcolors = {}
-- AABBGGRR
fontcolors['1'] = 'FFCAE2E9' -- title
fontcolors['2'] = 'FFCAE2E9' -- title highlight
fontcolors['3'] = 'FFCAE2E9' -- subtitle
fontcolors['4'] = 'FFEEFFFF' -- subtitle highlight
fontcolors['5'] = 'FF000000' -- parchment
fontcolors['6'] = 'FFEEFFFF' -- parchment highlight
fontcolors['7'] = 'FF000000' -- alternate parchment
fontcolors['8'] = 'FF102526' -- alternate parchment highlight
fontcolors['9'] = 'FFCDD6D9' -- semitrans dark background
fontcolors['A'] = 'FFFDE6E9' -- semitrans dark background highlight
fontcolors['B'] = 'FFCECECE'
fontcolors['C'] = 'FFC8F0C8' -- Minty update color
fontcolors['D'] = 'FF96C8FF' -- orangy subtitle color
fontcolors['E'] = 'FF808080' -- fake the unclickable colour
fontcolors['R'] = 'FF0000FF' -- red
fontcolors['G'] = 'FF00FF00' -- green
fontcolors['M'] = 'FF000080' -- parchment red
fontcolors['$'] = 'FF82E6FF' -- gold yellow/orange
```

Its clearly the color used for gold (`$`) so maybe not.

**Comment by bsdcode** [comment]
Thanks. I tried it, but it's too bright.

**Pull request merged**